cache/size_utils.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package cache
+
+import (
+	"math"
+	"reflect"
+)
+
+/*
+From src/runtime/hashmap.go:
```

Comment:
We are not using these structs; we just use them to estimate the map size. Having them here in the same file helps with understanding the code.